syntax = "proto3";

package vins.proto;

message Extrinsic {
  float x = 1;
  float y = 2;
  float z = 3;

  float qw = 4;
  float qx = 5;
  float qy = 6;
  float qz = 7;
}

message VinsConfig {
  // for camera intrinsic
  int32 image_width = 1;
  int32 image_height = 2;

  float fx = 3;
  float fy = 4;
  float cx = 5;
  float cy = 6;
  float k1 = 7;
  float k2 = 8;
  float p1 = 9;
  float p2 = 10;
  float k3 = 11;

  // for camera extrinsic : camera to imu
  Extrinsic camera_to_imu = 12;

  // imu intrinisic
  double acc_noise = 13;
  double acc_random_walk = 14;
  double gyr_noise = 15;
  double gyr_random_walk = 16;

  bool verbose = 17;
  bool viz = 18;
  bool equalize = 19;
  int32 max_num_pts = 20;
  int32 min_pt_distance = 21;
}
